app/src/main/java/com/onewelcome/showcaseapp/feature/userderegistration/UserDeregistrationScreen.kt

```diff
@@ -0,0 +1,164 @@
+        Constants.DOCUMENTATION_USER_DEREGISTRATION
+      )
+    },
+    settings = { SettingsSection(uiState, onEvent) },
+    result = uiState.result?.let { { DeregistrationResult(it) } },
+    action = { DeregisterUserButton(uiState, onEvent) }
+  )
+}
+
+@Composable
+private fun SettingsSection(uiState: State, onEvent: (UiEvent) -> Unit) {
+  Column(verticalArrangement = Arrangement.spacedBy(Dimensions.verticalSpacing)) {
+    SdkInitializationSection(uiState.isSdkInitialized)
+    RegisteredUserProfilesSection(uiState.registeredUserProfiles)
+    Text(
+      text = stringResource(R.string.required),
+      style = MaterialTheme.typography.titleSmall
+    )
+    UserProfileSelectionSection(uiState.registeredUserProfiles, uiState.selectedUserProfile, onEvent)
+  }
+}
+
+@Composable
+private fun SdkInitializationSection(isSdkInitialized: Boolean) {
+  ShowcaseStatusCard(
+    title = stringResource(R.string.status_sdk_initialized),
+    status = isSdkInitialized,
+    tooltipContent = { Text(stringResource(R.string.sdk_needs_to_be_initialized_to_perform_deregistration)) }
+  )
+}
+
+@Composable
+private fun RegisteredUserProfilesSection(registeredUserProfiles: Set<UserProfile>) {
+  val description = registeredUserProfiles.map { it.profileId }
+    .separateItemsWithComa()
+    .ifEmpty { stringResource(R.string.no_user_profiles) }
+  ShowcaseStatusCard(
+    title = stringResource(R.string.user_profiles),
+    description = description,
+    status = registeredUserProfiles.isNotEmpty(),
+    tooltipContent = { Text(stringResource(R.string.user_profiles_requirement_tooltip)) }
+  )
+}
+
+@Composable
+private fun UserProfileSelectionSection(
+  registeredUserProfiles: Set<UserProfile>,
+  selectedUserProfile: UserProfile?,
+  onEvent: (UiEvent) -> Unit
+) {
+  ShowcaseDropdownMenu(
+    label = {
+      Text(stringResource(R.string.label_user_profile))
+    },
+    itemList = registeredUserProfiles.toList(),
+    selectedItem = selectedUserProfile,
+    valueFormatter = { it.profileId },
+    tooltipContent = { Text(stringResource(R.string.documentation_choose_user_profile)) },
+    onItemSelected = { item, _ -> onEvent.invoke(UiEvent.OnUserProfileSelected(item)) }
+  )
+}
+
+@Composable
+private fun DeregistrationResult(result: Result<Unit, Throwable>) {
+  Column {
+    result.onSuccess { Text(stringResource(R.string.label_user_deregistration_success)) }
+      .onFailure {
+        val errorText = when (it) {
+          is OneginiError -> "${it.errorType.code}: ${it.message}"
+          else -> "$it"
+        }
+        Text(errorText)
+      }
+  }
+}
+
+@Composable
+private fun DeregisterUserButton(uiState: State, onEvent: (UiEvent) -> Unit) {
+  Button(
+    modifier = Modifier
+      .fillMaxWidth()
+      .height(Dimensions.actionButtonHeight),
+    onClick = { if (uiState.isLoading.not()) onEvent(UiEvent.DeregisterUser) },
+    enabled = uiState.selectedUserProfile != null
```

Comment:
We should not restrict users from hitting that button. If there's no selected user profile, SDK will throw some error that we should display. Same with SDK not initialized.